File: API/Helpers/UserParams.cs
﻿using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;

namespace API.Helpers;

public class UserParams : PaginationParams
{
    public string CurrentUserName { get; set; }
    public string Gender { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
}

File: client/src/app/_models/userParams.ts
import { IUser } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 12;
    orderBy = "lastActive";

    constructor(user: IUser) {
        this.gender = user.gender == 'female' ? 'male' : 'female';
    }
}